src/main/java/xyz/vegaone/easytrackingv2/mapper/UserStoryMapper.java

```diff
@@ -0,0 +1,44 @@
+package xyz.vegaone.easytrackingv2.mapper;
+
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import xyz.vegaone.easytrackingv2.domain.ProjectEntity;
+import xyz.vegaone.easytrackingv2.domain.UserStoryEntity;
+import xyz.vegaone.easytrackingv2.dto.Project;
+import xyz.vegaone.easytrackingv2.dto.UserStory;
+
+import java.util.List;
+
+@Mapper(componentModel = "Spring")
+public interface UserStoryMapper {
+
+    @Mapping(target = "project", ignore = true)
+    UserStory domainToDto(UserStoryEntity userStoryEntity);
+
+    @Mapping(target = "project", ignore = true)
+    UserStoryEntity dtoToDomain(UserStory userStory);
+
+    List<UserStory> domainToDtoList(List<UserStoryEntity> userStoryEntityList);
+
+    List<UserStoryEntity> dtoToDomain(List<UserStory> userStoryList);
+
+    @AfterMapping
+    default void addProjectToDto(UserStoryEntity userStoryEntity, @MappingTarget UserStory userStory) {
+        Project project = new Project();
```

Comment:
Thinking about it, we need a null check is here since I'm not sure that we will always include the `project` as part of the user story 
```java
if(userStoryEntity.getProject() != null) {
            Project project = new Project();
            project.setId(userStoryEntity.getProject().getId());
            project.setName(userStoryEntity.getProject().getName());

            userStory.setProject(project);
        }
```